File: packages/app/src/services/software/definitions/variants/discordStable.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'discord-stable';
	public name = 'Discord Stable';
	public category = [SoftwareCategory.Social];
	public downloadName = 'DiscordSetup.exe';
	public icon = 'discord.png';
	public homepage = 'https://discord.com';
	public parent = 'discord';

	public async resolveDownloadUrl() {
		return ok('https://discord.com/api/downloads/distributions/app/installers/latest?channel=stable&platform=win&arch=x64');
	}
}

File: src/lib/software/variants/discord_stable.py
from src.lib.software import BaseSoftware

class DiscordStable(BaseSoftware):
    def __init__(self):
        super().__init__()

        self.download_name = 'DiscordSetup.exe'
        self.silent_install_args = ['--silent']

    def resolve_download_url(self):
        self.url_resolved.emit('https://discord.com/api/downloads/distributions/app/installers/latest?channel=stable&platform=win&arch=x64')
